### ISSUE TITLE
Endless fetching messages (>10 minutes)

### ISSUE THREAD

**Issue by annadanchenko** [status-im/status-react#7679]
### Description
[comment]: # (Feature or Bug? i.e Type: Bug)
_Type_: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
_Summary_: Fetching indicator is moving for > 10 minutes. First ~5 minutes I've joined no chats, later I've joined #testanna chat and still see no history after waiting for ~8 minutes. After restarting the app history for chat is retrieved in less then 5 sec. So, it looks like issue is with the first launch.

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
Fetching indicator stops

#### Actual behavior
[comment]: # (Describe what actually happened.)
Moving fetching indicator when

1. no chats are joined
1. when chat is joined and waited for 8 minutes
### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

* Install Status and give the app Storage permission
* Create account, change log level to Info
* log in to the app (turn on saving password)
* open Profile screen
* return to home screen and wait till Fetching indicator will stop moving

### Additional Information
[comment]: # (Please do your best to fill this out.)

* Status version: 0.10.1 build 6 (hotfix release candidate)
* Operating System: Android 9, real Pixel XL

#### Logs when indicator was moving for > 20 minutes
[geth.log](https://github.com/status-im/status-react/files/2930967/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2930968/Status.log)

```
{
"client": {
"app-version": "0.10.1 (2019030417)",
"mailserver": "mail-02.ac-cn-hongkong-c.eth.beta",
"node-info": {
"id": "9f562c8025fca09fca521764be60a80f94bce1f1f82024f13d6e920b47133896",
"name": "StatusIM/v0.23.0-beta.7/android-arm/go1.11",
"enode": "enode://74eb25d43959e413b02188d770def22c81734642a238c4664f5c5423f4ddab7d081cc7e7f3f2c7384bf63ac52b901f8826849252a194d4c6c5fe5d7cd250fbf5@127.0.0.1:48968?discport=0",
"enr": "0xf88fb8408ec9415f28eb49ba7b56ae9e8a57a6036495b0052a950756cb74b3345351a19e6f402099c90627486f3846c0438324ab888bda6e4af835e6d6f60f58faaeb0940283636170c6c58373686806826964827634826970847f00000189736563703235366b31a10374eb25d43959e413b02188d770def22c81734642a238c4664f5c5423f4ddab7d8374637082bf48",
"ip": "127.0.0.1",
"ports": {
"discovery": 0,
"listener": 48968
},
"listenAddr": "[::]:48968",
"protocols": {
"shh": {
"maxMessageSize": 1048576,
"minimumPoW": 0.001,
"version": "6.0"
}
}
},
"node-version": "status-go v0.23.0-beta.7"
},
"peers": [
{
"enode": "enode://19872f94b1e776da3a13e25afa71b47dfa99e658afd6427ea8d6e03c22a99f13590205a8826443e95a37eee1d815fc433af7a8ca9a8d0df7943d1f55684045b7@35.238.60.236:30305?discport=0",
"id": "10841e6db5c02fc331bf36a8d2a9137a1696d9d3b6b1f872f780e02aa8ec5bba",
"name": "Statusd/v0.23.0-beta.6/linux-amd64/go1.10.2",
"caps": [
"shh/6"
],
"network": {
"localAddress": "192.168.178.26:39075",
"remoteAddress": "35.238.60.236:30305",
"inbound": false,
"trusted": false,
"static": true
},
"protocols": {
"shh": "unknown"
}
},
{
"enode": "enode://4ea35352702027984a13274f241a56a47854a7fd4b3ba674a596cff917d3c825506431cf149f9f2312a293bb7c2b1cca55db742027090916d01529fe0729643b@206.189.243.178:443",
"id": "40a8cfd21a3a2a45ff6b18c6db3ba5dfe4acd8d4be6c17e84554d648365e7547",
"name": "Statusd/v0.23.0-beta.6/linux-amd64/go1.10.2",
"caps": [
"shh/6"
],
"network": {
"localAddress": "192.168.178.26:40849",
"remoteAddress": "206.189.243.178:443",
"inbound": false,
"trusted": false,
"static": true
},
"protocols": {
"shh": "unknown"
}
},
{
"enode": "enode://744098ab6d3308af5cd03920aea60c46d16b2cd3d33bf367cbaf1d01c2fcd066ff8878576d0967897cd7dbb0e63f873cc0b4f7e4b0f1d7222e6b3451a78d9bda@47.89.20.15:443",
"id": "74b75499826d6fa03d17bd2e15a255b7a50b1c4267e017f033aa4906374b440e",
"name": "Statusd/v0.23.0-beta.6/linux-amd64/go1.10.2",
"caps": [
"shh/6"
],
"network": {
"localAddress": "192.168.178.26:40973",
"remoteAddress": "47.89.20.15:443",
"inbound": false,
"trusted": false,
"static": true
},
"protocols": {
"shh": "unknown"
}
},
{
"enode": "enode://b33dc678589931713a085d29f9dc0efee1783dacce1d13696eb5d3a546293198470d97822c40b187336062b39fd3464e9807858109752767d486ea699a6ab3de@35.193.151.184:30305?discport=0",
"id": "863a31b6c5eeed60c017d2e1247db134c02976bc96a5caf65a009b788e391057",
"name": "Statusd/v0.23.0-beta.6/linux-amd64/go1.10.2",
"caps": [
"shh/6"
],
"network": {
"localAddress": "192.168.178.26:48026",
"remoteAddress": "35.193.151.184:30305",
"inbound": false,
"trusted": false,
"static": true
},
"protocols": {
"shh": "unknown"
}
},
{
"enode": "enode://aa97756bc147d74be6d07adfc465266e17756339d3d18591f4be9d1b2e80b86baf314aed79adbe8142bcb42bc7bc40e83ee3bbd0b82548e595bf855d548906a1@47.52.188.241:443",
"id": "a681612c0a4b3321926dfe00af7bb5891d07a512a24b5ba6b05e11703d66aec3",
"name": "Statusd/v0.23.0-beta.6/linux-amd64/go1.10.2",
"caps": [
"shh/6"
],
"network": {
"localAddress": "192.168.178.26:47935",
"remoteAddress": "47.52.188.241:443",
"inbound": false,
"trusted": false,
"static": true
},
"protocols": {
"shh": "unknown"
}
}
]
}
```

# Logs after restarting app (when history is retrieved fine)
[geth.log](https://github.com/status-im/status-react/files/2930997/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2931000/Status.log)

**Comment by annadanchenko** [comment]
tried again on new installation, hotfix build 26. 
Fetching stopped after 8 minutes. No chats were joined.

[geth.log](https://github.com/status-im/status-react/files/2931077/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2931078/Status.log)

**Comment by Serhy** [comment]
It was all OK with two new accounts I created (Android 8.1) and fetching blue indicator finished in 1-1.5 mins...
However third account I created and without joining any chat - during next 12 minutes 'Fetching messages' blue indicator was still present. During that time mailserver changed only 1 time (from `mail-02.ac-cn-hongkong` to `mail-01.ac-cn-hongkong`) but several times it was visible mailserver reconnecting (the green 'Connected' label on Home screen). 
Below are logs starting from this affected account creation:

[geth.log](https://github.com/status-im/status-react/files/2931088/geth.44.log) - Debug Level
[logcat.log](https://github.com/status-im/status-react/files/2931089/logcatendlessconnec.log)

After app restart and logging into this account it connected fine (stopped on `mail-02.gc-us-central1` mailsever)

**Comment by annadanchenko** [comment]
on @cammellos request: checked on 0.10.0 when mail-01.ac-cn-hongkong is selected upon app launch, after this joined #status channel and got history in ~2 min. 

[geth.log](https://github.com/status-im/status-react/files/2931170/geth.log)
[Status.log](https://github.com/status-im/status-react/files/2931171/Status.log)

**Comment by asemiankevich** [comment]
i can't now use any mailserver (literally) on iOS, even same mailserver works fine on Android devices.  Version 0.10.1 , go 0.23.0-beta.7

@annadanchenko @cammellos

**Comment by cammellos** [comment]
@asemiankevich do you have any logs for that session? thanks!

**Comment by asemiankevich** [comment]
> @asemiankevich do you have any logs for that session? thanks!

nope :(

**Comment by annadanchenko** [comment]
was fixed in hotfix 0.10.1 release build 30 by reverting "Fix concurrent retry-request". Still present in develop

**Comment by annadanchenko** [comment]
works fine in last release 0.13.0
